examples/step-75/doc/intro.dox

```diff
@@ -0,0 +1,230 @@
+number of degrees of freedom assigned, and approximately the same
+workload. To balance the workload among all processes, we have to
+balance the number of cells on all participating processes.
+
+With hp-adaptive methods, this is no longer the case: the finite element
+type may vary from cell to cell and consequently also the number of
+degrees of freedom. Matching the number of cells does not yield a
+balanced workload. In the matrix-free context, the workload can be
+assumed to be proportional the number of degrees of freedom of each
+process, since in the best case only the source and the destination
+vector have to be loaded.
+
+One could balance the workload by assigning weights to every cell which
+are proportional to the number of degrees of freedom, and balance the
+sum of all weights between all processes. Assigning individual weights
+to each cell can be realized with the class parallel::CellWeights which
+we will use later.
+
+
+<h3>hp-decision indicators</h3>
+
+With hp-adaptive methods, we not only have to decide which cells we want
+to refine or coarsen, but we also have the choice how we want to do
+that: either by adjusting the grid resolution or the polynomial degree
+of the finite element.
+
+We will again base the decision on which cells to adapt on (a
+posteriori) computed error estimates of the current solution, e.g.,
+using the KellyErrorEstimator. We will similarly decide how to adapt
+with (a posteriori) computed smoothness estimates: large polynomial
+degrees work best on smooth parts of the solution while fine grid
+resolutions are favorable on irregular parts. In step-27, we presented a
+way to calculate smoothness estimates based on the decay of Fourier
+coefficients. Let us take the opportunity and present an alternative
+that follows the same idea, but with Legendre coefficients.
+
+We will briefly present the idea of this new technique, but limit its
+description to 1D for simplicity. A general function can be represented
+with Legendre polynomials as basis functions:
+@f[
+u(x) = \sum\limits_{k \geq 0} c_k P_k(x) ,
+@f]
+and the Legendre expansion coefficients can be calculated as follows:
+@f[
+c_k = \frac{2k+1}{2} \int u(x) P_k(x) dx .
+@f]
+The first factor is a normalization constant. With a
+mapping $F_K$ from an actual cell $K$ to the reference cell
+$\hat{K}$, we can get the coefficients for our finite element
+approximation $u_\text{hp}$ while performing all calculations on the
+reference cell:
+@f[
+c_{k,K} = \frac{2k+1}{2} \int\limits_{\hat K} \hat{u}_\text{hp}(\hat{x})
+\hat{P}_k(\hat{x}) \left|\det \mathbf{J}(\hat{x})\right|
+\mathrm{d}\hat{x} ,
+@f]
+where $\varphi(x) = \varphi(F_K(\hat{x})) = \hat{\varphi}(\hat{x})$ for
+a general function $\varphi$. A non-degenerate mapping, i.e.,
+$\left|\det \mathbf{J}(\hat{x})\right| \approx 1$, allows us to perform
+the transformation on any cell $K$ exclusively on the reference cell
+$\hat{K}$ and pre-calculate the necessary transformation matrices only
+once:
```

Comment:
I'm not sure this is right. If `J` is `1` or nearly `1` then the mapping is a translation or rotation, but this is basically never the case for finite elements. Is this described in one of Melenk's papers or some other citation?

Reply:
You are right, this is basically never the case for finite elements and poses a weakness of the current `FESeries` classes. It would be too expensive to build such a transformation matrix with the correct mapping on every single cell. The same logic is applied on the [old version of the `step-27` tutorial](https://www.dealii.org/8.3.0/doxygen/deal.II/step_27.html#LaplaceProblemestimate_smoothness) where this generalization stems from.

We should be more clear here and say that this only works correctly with a non-degenerate mapping, and that results on distorted cells are inaccurate.

Reply:
Does that mean that the code works correctly only if the Jacobian is constant?

Reply:
Yes, the Fourier and Legendre transformations are only correct for constant Jacobians at the moment.
We only estimate the smoothness with these transformations, so this only impacts the choice of the refinement type (h or p). I don't consider this to be too critical.

Reply:
Ah, okay. We should clarify this text to say that this code is only correct for affine cells (not `J == 1`) then.

To be clear, I'm not worried about the functionality here or in step-27, I just want to make sure I (and anyone else reading this) understand what the condition on the cell geometry (i.e., the properties of the mapping back to the unit cell) is.